spring-boot-project/spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/Layout.java

```diff
@@ -49,6 +49,12 @@
 	 */
 	String getClassesLocation();
 
+	/**
+	 * Returns the location of the libraries within the archive.
+	 * @return the libraries location
+	 */
+	String getLibrariesLocation();
```

Comment:
There's a lot of overlap with `getLibraryDestination(String, LibraryScope)`. Without the `LibraryScope`, this method will return the wrong location for a provided artifact in a war – it will return `WEB-INF/lib` when it should return `WEB-INF/lib-provided`. I wonder if the proposed solution can be reworked to use `getLibraryDestination(String, LibraryScope)` instead?